File: Programmers/Algorithm_Kit/05_brute_force/001.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int row, col;

int solution(vector<vector<int>> sizes) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    
    for(vector<int> sz : sizes) {
        if (sz[0] < sz[1]) {
            swap(sz[0], sz[1]);
        }
        
        row = max(row, sz[0]);
        col = max(col, sz[1]);
    }
    
    int answer = row * col;
    
    return answer;
}

File: programmers/algorithmkit/05_brute_force/001.java
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;

        int row_max = -1;
        int col_max = -1;

        for (int[] s : sizes) {
            if (s[0] < s[1]) {
                int temp = s[0];
                s[0] = s[1];
                s[1] = temp;
            }

            row_max = Math.max(row_max, s[0]);
            col_max = Math.max(col_max, s[1]);
        }

        answer = row_max * col_max;

        return answer;
    }
}
